### PR TITLE
[Calling][Feature] Features/rooms flagging + SwiftLint 0.53 

### PR DESCRIPTION
**Pull request by ahammer**
Flagging for rooms

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift [hunk 1]
```diff
@@ -247,8 +247,7 @@ extension ChatDemoView {
             ChatCompositeErrorCode.fetchMessagesFailed,
             ChatCompositeErrorCode.requestParticipantsFetchFailed,
             ChatCompositeErrorCode.sendReadReceiptFailed,
-            ChatCompositeErrorCode.sendTypingIndicatorFailed,
-            ChatCompositeErrorCode.disconnectFailed:
```
Comment (pavelprystinka) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift:251]: It is accidentally removed?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallHistoryRecord.swift [hunk 2]
```diff
@@ -13,9 +13,4 @@ public struct CallHistoryRecord {
 
     /// Call Ids associated with particular call started on callStartedOn date.
     public let callIds: [String]
-
-    init(callStartedOn: Date, callIds: [String]) {
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallHistoryRecord.swift:17]: Why do we need to remove init?

**Review by pavelprystinka — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift (hunk 1 above)
Comment (ahammer) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift:251]: See above

        case ChatCompositeErrorCode.joinFailed:
            errorMessage = "Connection Failed"
        case ChatCompositeErrorCode.disconnectFailed:
            errorMessage = "Disconnect Failed"
            
         They were duplicates in the case, dead branches.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallHistoryRecord.swift (hunk 2 above)
Comment (ahammer) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallHistoryRecord.swift:17]: synthetic constructor, with same order as params, swift automatically generates, swiftlint says to delete.

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift (hunk 1 above)
Comment (ahammer) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift:251]: duplicate cases, caught above.

**Review by ShaunaSong — APPROVED** [review]

**Review by jamchengms — APPROVED** [review]

File: AzureCommunicationUI/.swiftlint.yml [hunk 3]
```diff
@@ -99,7 +99,7 @@ opt_in_rules:
   - redundant_string_enum_value
 
   # Properties, variables, and constants should not have redundant type annotation
-  - redundant_type_annotation
+  # - redundant_type_annotation
```
Comment (emlynmac) [inline AzureCommunicationUI/.swiftlint.yml:102]: Why disable this?

File: AzureCommunicationUI/.swiftlint.yml [hunk 4]
```diff
@@ -190,9 +192,10 @@ modifier_order:
 type_name:
   min_length: 3
   max_length: 50
+
 function_body_length:
-  - 75 # Warning
-  - 100 # Error
+  - 150 # Warning
```
Comment (emlynmac) [inline AzureCommunicationUI/.swiftlint.yml:197]: Long functions discouraged; less than a screen ideally.

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Model/ReadReceiptInfoModel.swift [hunk 5]
```diff
@@ -12,12 +12,6 @@ struct ReadReceiptInfoModel: BaseInfoModel, Equatable {
     let chatMessageId: String
     let readOn: Iso8601Date
 
-    init(senderIdentifier: CommunicationIdentifier, chatMessageId: String, readOn: Iso8601Date) {
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Model/ReadReceiptInfoModel.swift:15]: Why is this removed?

Comment (ahammer) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Model/ReadReceiptInfoModel.swift:15]: swift synthetic constructors on struct, it's generated 1:1 for you

swiftlint said to delete since not needed.

File: AzureCommunicationUI/.swiftlint.yml (hunk 3 above)
Comment (ahammer) [reply AzureCommunicationUI/.swiftlint.yml:102]: We don't follow it, have 100s of warnings on it in swiftlint for it, that drown out other issues.

it could be re-enabled because they aren't errors, but then I have to chase stamps again.

File: AzureCommunicationUI/.swiftlint.yml (hunk 4 above)
Comment (ahammer) [reply AzureCommunicationUI/.swiftlint.yml:197]: I agree, but these are tight and get over-written in code, and swiftlint wants other updates to files to over-ride (not allowing blanket over-rides anymore).

If this is a deal breaker, I'll need to split this up because I need to chase to many people to PR this.

**Review by vhuseinova-msft — APPROVED** [review]

**Review by JoshuaLai — APPROVED** [review]
New work item to revert the changes in the swiftlint will be needed. Approving for now to unblock PR
